src/components/Calculator.js

```diff
@@ -1,42 +1,54 @@
+import { useState } from 'react';
 import styles from '../styles/Calculator/Calculator.module.css';
 import Button from './Button';
 import ResultBar from './ResultBar';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const [state, setState] = useState({});
+
+  const displayValue = state.next || state.total || '0';
+
+  const handleClick = (value) => {
+    const newState = calculate(state, value);
+    setState(newState);
+  };
+
   return (
     <section>
       <div className={styles.calculator}>
-        <ResultBar />
+        <ResultBar displayValue={displayValue} />
         <div className={styles.row}>
-          <Button value="AC" />
-          <Button value="+/-" />
-          <Button value="%" />
-          <Button value="÷" accent />
+          <Button value="AC" handleClick={handleClick} />
+          <Button value="+/-" handleClick={handleClick} />
+          <Button value="%" handleClick={handleClick} />
+          <Button value="÷" accent handleClick={handleClick} />
         </div>
         <div className={styles.row}>
-          <Button value="7" />
-          <Button value="8" />
-          <Button value="9" />
-          <Button value="x" accent />
+          <Button value="7" handleClick={handleClick} />
+          <Button value="8" handleClick={handleClick} />
+          <Button value="9" handleClick={handleClick} />
+          <Button value="x" accent handleClick={handleClick} />
         </div>
         <div className={styles.row}>
-          <Button value="4" />
-          <Button value="5" />
-          <Button value="6" />
-          <Button value="-" accent />
+          <Button value="4" handleClick={handleClick} />
+          <Button value="5" handleClick={handleClick} />
+          <Button value="6" handleClick={handleClick} />
+          <Button value="-" accent handleClick={handleClick} />
         </div>
         <div className={styles.row}>
-          <Button value="1" />
-          <Button value="2" />
-          <Button value="3" />
-          <Button value="+" accent />
+          <Button value="1" handleClick={handleClick} />
+          <Button value="2" handleClick={handleClick} />
+          <Button value="3" handleClick={handleClick} />
+          <Button value="+" accent handleClick={handleClick} />
         </div>
         <div className={styles.row}>
-          <Button value="0" wide />
-          <Button value="." />
-          <Button value="=" accent />
+          <Button value="0" wide handleClick={handleClick} />
+          <Button value="." handleClick={handleClick} />
+          <Button value="=" accent handleClick={handleClick} />
         </div>
       </div>
+      <pre>{JSON.stringify(state, null, 2)}</pre>
```

Comment:
- [ Optional ] I think it would be better if you remove this pre-formatted text from your project, I think it serves you best when you are debugging the application but in production, users might find it to be very cryptic. Kindly see the screenshot below and rectify this.

## Screenshot
![Screenshot from 2023-07-13 00-25-18](https://github.com/henokkhm/microverse-math-magicians/assets/16392046/926b54bd-ec86-4965-a49a-8634e9ddfc1e)

Reply:
@Felix45 Thanks for the suggestion. I have removed it.